tests/src/test/java/org/eclipse/hono/tests/http/TelemetryHttpIT.java

```diff
@@ -194,8 +199,22 @@ public void testUploadQos1MessageFailsIfDeliveryStateNotUpdated(final VertxTestC
                         tenantId, 
                         (delivery, msg) -> {
                             logger.debug("received {}", msg);
+                            ctx.verify(() -> {
+                                assertThat(delivery.remotelySettled()).isFalse();
+                                assertThat(delivery.getRemoteState()).isNull();
+                            });
                             messageReceived.flag();
                             // don't update the delivery state here
+                            // wait for the delivery to get remotely settled via the timeout handling in the adapter
+                            vertx.setTimer(ClientConfigProperties.DEFAULT_SEND_MESSAGE_TIMEOUT + 50, tid -> {
```

Comment:
IMHO we should tie this check to *after* the HTTP client has gotten back its 501 response (and then wait some) in order to make it more robust against timing issues induced by scarce computing resources on the CI system ...